resources/aws/securitygroup.go

```diff
@@ -15,8 +19,47 @@ type SecurityGroup struct {
 	AWSEntity
 }
 
+func (s *SecurityGroup) existingSecurityGroupID() (string, error) {
+	securityGroups, err := s.Clients.EC2.DescribeSecurityGroups(&ec2.DescribeSecurityGroupsInput{
+		Filters: []*ec2.Filter{
+			&ec2.Filter{
+				Name: aws.String("description"),
+				Values: []*string{
+					aws.String(s.Description),
+				},
+			},
+			&ec2.Filter{
+				Name: aws.String("group-name"),
+				Values: []*string{
+					aws.String(s.GroupName),
+				},
+			},
+		},
+	})
+	if err != nil {
+		return "", microerror.MaskAny(err)
+	}
+
+	for _, securityGroup := range securityGroups.SecurityGroups {
+		return *securityGroup.GroupId, nil
+	}
+
+	return "", microerror.MaskAny(securityGroupCreateAndFindError)
+}
+
 func (s *SecurityGroup) CreateIfNotExists() (bool, error) {
-	return false, microerror.MaskAny(notImplementedMethodError)
+	if err := s.CreateOrFail(); err != nil {
+		if strings.Contains(err.Error(), awsclient.SecurityGroupDuplicate) {
+			id, err := s.existingSecurityGroupID()
+			if err != nil {
+				return false, microerror.MaskAny(err)
+			}
+			s.id = id
+			return false, nil
+		}
+		return false, microerror.MaskAny(err)
+	}
+	return true, nil
```

Comment:
I think it would be nice to have newlines before returns here and on line 60, for consistency.